glu.go

```diff
@@ -184,7 +239,16 @@ func (s *System) Run() error {
 		shutdownCtx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
 		defer cancel()
 
-		return srv.Shutdown(shutdownCtx)
+		// call in reverse order to emulate pop semantics of a stack
+		for i := len(s.shutdownFuncs) - 1; i >= 0; i-- {
+			if fn := s.shutdownFuncs[i]; fn != nil {
+				if err := fn(shutdownCtx); err != nil {
+					slog.Error("shutting down", "error", err)
+				}
+			}
+		}
```

Comment:
Take it or leave it:
```suggestion
		for _, fn := slices.Backward(s.shutdownFuncs) {
			if err := fn(shutdownCtx); err != nil {
				slog.Error("shutting down", "error", err)
			}
		}
```